src/unity/python/turicreate/toolkits/drawing_recognition/drawing_recognition.py

```
@@ -0,0 +1,480 @@
+from turicreate.toolkits import evaluation as _evaluation
+import turicreate.toolkits._internal_utils as _tkutl
+from ._sframe_loader import SFrameRecognitionIter as _SFrameRecognitionIter
+from .. import _mxnet_utils
+from turicreate import extensions as _extensions
+from copy import copy as _copy
+
+BITMAP_WIDTH = 28
+BITMAP_HEIGHT = 28
+
+class Model(_HybridBlock):
+    def __init__(self, num_classes, **kwargs):
+        super(Model, self).__init__(**kwargs)
+        with self.name_scope():
+            # layers created in name_scope will inherit name space
+            # from parent layer.
+            self.conv1 = _nn.Conv2D(channels=16, kernel_size=(3,3), 
+                                    padding=(1,1), activation='relu')
+            self.pool1 = _nn.MaxPool2D(pool_size=(2,2))
+            self.conv2 = _nn.Conv2D(channels=32, kernel_size=(3,3), 
+                                    padding=(1,1), activation='relu')
+            self.pool2 = _nn.MaxPool2D(pool_size=(2,2))
+            self.conv3 = _nn.Conv2D(channels=64, kernel_size=(3,3), 
+                                    padding=(1,1), activation='relu')
+            self.pool3 = _nn.MaxPool2D(pool_size=(2,2))
+            self.flatten = _nn.Flatten()
+            self.fc1 = _nn.Dense(units=128, activation='relu')
+            self.fc2 = _nn.Dense(units=num_classes, activation=None)
+
+    def hybrid_forward(self, F, x):
+        x = self.pool1(self.conv1(x))
+        x = self.pool2(self.conv2(x))
+        x = self.pool3(self.conv3(x))
+        x = self.flatten(x)
+        x = self.fc1(x)
+        x = self.fc2(x)
+        return F.softmax(x)
+
+def create(input_dataset, annotations=None, num_epochs=100, feature="bitmap", model=None,
+           classes=None, batch_size=256, max_iterations=0, verbose=True, 
+           **kwargs):
+    """
+    Create a :class:`DrawingRecognition` model.
+    """
+    start_time = _time.time()
+
+    is_stroke_input = (input_dataset[feature].dtype != _tc.Image)
```

Should we throw an error if the input is neither images nor stroke data? What if you an SArray of integers or something?

---

Absolutely. We should throw a ToolkitError.